lib/wind/utf8.c

```diff
@@ -87,9 +97,16 @@ utf8toutf32(const unsigned char **pp, uint32_t *out)
 	    return WIND_ERR_INVALID_UTF8;
 	}
     } else {
-	*out = c;
+	out_val = c;
     }
 
+    /* Allow unpaired surrogates (in the range 0xd800–0xdfff). */
```

Comment:
Do we have to?

Reply:
Yeah, I thought we might want to continue to allow unpaired surrogates in case something depends on their use. I’ve known Windows to produce them, for example.